### PR TITLE
DRYD-1358: Update Held-in-Trust

### PR DESCRIPTION
**Pull request by mikejritter**
**What does this do?**
* Update Held-in-Trust for NAGPRA workflows

**Why are we doing this? (with JIRA link)**
Jira: https://collectionspace.atlassian.net/browse/DRYD-1358

This updates the Held-in-Trust procedure which was going to be added for 8.0 to be compatible with NAGPRA related workflows. It 

**How should this be tested? Do these changes have associated tests?**
* Run `npm run lint` and `npm run tests`
* Rebuild collectionspace with the core tenant enabled
* Start collectionspace
* Start the devserver `npm run devserver`
* Check that the Held-in-Trust procedure saves and loads as expected
  * With the exception of heldintrust type which needs default terms
* Optionally check the messages in the advanced search

**Dependencies for merging? Releasing to production?**
None

**Has the application documentation been updated for these changes?**
No

**Did someone actually run this code to verify it works?**
@mikejritter tested against a local instance

### CODE REVIEW

File: src/plugins/recordTypes/heldintrust/fields.js [hunk 1]
```diff
@@ -60,157 +60,58 @@ export default (configContext) => {
...
-                  },
-                  name: {
-                    id: 'field.heldintrusts_common.depositorNote.name',
-                    defaultMessage: 'Note',
-                  },
-                }),
-                view: {
-                  type: TextInput,
-                },
+              type: AutocompleteInput,
```
Comment (ray-lee) [inline src/plugins/recordTypes/heldintrust/fields.js:72]: Looks like this should be TermPickerInput.

File: src/plugins/recordTypes/heldintrust/fields.js
Comment (ray-lee) [inline src/plugins/recordTypes/heldintrust/fields.js:1]: Should we add the unique identification number validator to this? I think that didn't exist when this was originally implemented.

**Review by ray-lee — CHANGES_REQUESTED** [review]

File: src/plugins/recordTypes/heldintrust/fields.js
Comment (mikejritter) [reply src/plugins/recordTypes/heldintrust/fields.js:1]: Oh good call, I overlooked that when bringing in Held-in-Trust previously.

**Review by ray-lee — APPROVED** [review]
Looks good!

**Pull request merged**